tests/test_init.py

```diff
@@ -351,3 +351,53 @@ def test_zero_wires(self, init, sgntr):
         p = init(**sgntr)
 
         assert p.flatten().shape == (0,)
+
```

Comment:
Similar to #875, it would be good to add a short test to ensure that the decomposition of U2 is correct, mostly since we have to be careful not to blindly trust the results of the paper. 

You can take a look at the recently merged test `test_decomposition_u1ex` and replicate it for the `CNOT` `CRX` `CNOT` decomposition of the U2 exchange.